### PR TITLE
PWX-34441: Integration tests for excludeResourceType parameter.

### PR DESCRIPTION
**Pull request by diptiranjanpx**
**What type of PR is this?**
integration-test

**What this PR does / why we need it**:
Tests for the newly added excludeResourceType parameter.

**Does this PR change a user-facing CRD or CLI?**:

**Is a release note needed?**:

**Does this change need to be cherry-picked to a release branch?**:

**Test**
`
Ran the tests locally and following is the result.
--- PASS: TestMigration (2742.82s)
    --- PASS: TestMigration/testMigration (2661.95s)
        --- PASS: TestMigration/testMigration/excludeResourceTypeTest (461.22s)
        --- PASS: TestMigration/testMigration/excludeResourceTypePVCTest (437.38s)
        --- PASS: TestMigration/testMigration/excludeMultipleResourceTypesTest (416.45s)
        --- PASS: TestMigration/testMigration/excludeResourceTypesWithSelectorsTest (439.33s)
        --- PASS: TestMigration/testMigration/excludeNonExistingResourceTypesTest (900.32s)
    --- PASS: TestMigration/testMigrationFailoverFailback (77.17s)
PASS

DONE 8 tests in 2850.503s
latest run:
https://jenkins.pwx.dev.purestorage.com/job/Users/job/dipti/job/diptistorkcustom/36/
`

### CODE REVIEW

File: test/integration_test/migration_test.go [hunk 1]
```diff
@@ -2027,6 +2032,288 @@ func transformResourceTest(t *testing.T) {
...
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout/2, defaultWaitInterval)
+	require.NoError(t, err, "Migration could not be completed")
+
+	// Change kubeconfig to destination
+	err = setDestinationKubeConfig()
+	require.NoError(t, err, "failed to set kubeconfig to source cluster: %v", err)
+
+	destDeployments, err := apps.Instance().ListDeployments(namespace, meta_v1.ListOptions{})
+	require.NoError(t, err, "error retriving deployments list from %s namespace", namespace)
+	require.Equal(t, 1, len(destDeployments.Items), fmt.Sprintf("Expected 1 deployment in destination in %s namespace", namespace))
```
Comment (pp511) [inline test/integration_test/migration_test.go:2227]: Shouldn't we have PVC in this list too?

File: test/integration_test/specs/exclude-resourcetype-migration/migration.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  # If set to false this will migrate only the volumes. No PVCs, apps, etc will be migrated
+  includeResources: true
+  # If set to false, the deployments and stateful set replicas will be set to 0 on the destination. There will be an annotation with "stork.openstorage.org/migrationReplicas" to store the replica count from the source
+  startApplications: true
+  # 
+  excludeResourceTypes:
+  - Deployment
+  # List of namespaces to migrate
+  namespaces:
+  - mysql-1-pvc-exclude-resourcetype-migration
```
Comment (olavangad-px) [inline test/integration_test/specs/exclude-resourcetype-migration/migration.yaml:17]: nit: 
I think the namespace should be mysql-1-pvc-exclude-resourcetype-deployment

File: test/integration_test/migration_test.go [hunk 3]
```diff
@@ -2027,6 +2032,288 @@ func transformResourceTest(t *testing.T) {
...
+	require.NoError(t, err, "failed to set kubeconfig to source cluster: %v", err)
+	destroyAndWait(t, ctxs)
+
+	blowNamespacesForTest(t, instanceID, appKey, false)
+	// If we are here then the test has passed
+	testResult = testResultPass
+	logrus.Infof("Test status at end of %s test: %s", t.Name(), testResult)
+}
+
+func excludeNonExistingResourceTypesTest(t *testing.T) {
```
Comment (olavangad-px) [inline test/integration_test/migration_test.go:2244]: nit: What do you mean by NonExistingResourceType here ??
To me it sounds like we put an invalid resource type in which case Migration actually fails

Comment (diptiranjanpx) [reply test/integration_test/migration_test.go:2244]: It's like there are no statefulsets in the namespace but we want to exclude statefulsetype in migration

File: test/integration_test/migration_test.go (hunk 1 above)
Comment (olavangad-px) [reply test/integration_test/migration_test.go:2227]: +1

File: test/integration_test/migration_test.go [hunk 4]
```diff
@@ -2027,6 +2032,288 @@ func transformResourceTest(t *testing.T) {
...
+	testResult = testResultPass
+	logrus.Infof("Test status at end of %s test: %s", t.Name(), testResult)
+}
+
+func excludeMultipleResourceTypesTest(t *testing.T) {
+	var testrailID, testResult = 93404, testResultFail
+	runID := testrailSetupForTest(testrailID, &testResult)
+	defer updateTestRail(&testResult, testrailID, runID)
+	instanceID := "exclude-multiple-resourcetypes-migration"
+	appKey := "mysql-1-pvc"
```
Comment (olavangad-px) [inline test/integration_test/migration_test.go:2142]: I can see that mysql-1-pvc will schedule a pvc,deployment and service in the source cluster
So we should also validate that PVC was migrated in this case

Comment (olavangad-px) [reply test/integration_test/migration_test.go:2142]: Same thing in other places?
The resource type which is not excluded we should validate was successfully migrated?

File: test/integration_test/migration_test.go [hunk 5]
```diff
@@ -2027,6 +2032,288 @@ func transformResourceTest(t *testing.T) {
...
+		false,
+		"",
+		nil)
+
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout/2, defaultWaitInterval)
+	require.NoError(t, err, "Migration could not be completed")
+
+	// Change kubeconfig to destination
+	err = setDestinationKubeConfig()
+	require.NoError(t, err, "failed to set kubeconfig to source cluster: %v", err)
```
[the remark this answers is not in the document]
Comment (diptiranjanpx) [reply test/integration_test/migration_test.go]: Done

File: test/integration_test/migration_test.go (hunk 1 above)
Comment (diptiranjanpx) [reply test/integration_test/migration_test.go:2244]: yes, added the check for other resources.

File: test/integration_test/specs/exclude-resourcetype-migration/migration.yaml (hunk 2 above)
Comment (diptiranjanpx) [reply test/integration_test/specs/exclude-resourcetype-migration/migration.yaml]: done

File: test/integration_test/migration_test.go (hunk 4 above)
Comment (diptiranjanpx) [reply test/integration_test/migration_test.go:2154]: done

File: test/integration_test/migration_test.go [hunk 6]
```diff
@@ -2027,6 +2032,288 @@ func transformResourceTest(t *testing.T) {
...
+
+	destDeployments, err := apps.Instance().ListDeployments(namespace, meta_v1.ListOptions{})
+	require.NoError(t, err, "error retriving deployments list from %s namespace", namespace)
```
[the remark this answers is not in the document]
Comment (diptiranjanpx) [reply test/integration_test/migration_test.go]: done

File: test/integration_test/migration_test.go [hunk 7]
```diff
@@ -2027,6 +2032,288 @@ func transformResourceTest(t *testing.T) {
...
+		false,
+		"",
+		nil)
+
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout/2, defaultWaitInterval)
+	require.NoError(t, err, "Migration could not be completed")
+
+	// Change kubeconfig to destination
+	err = setDestinationKubeConfig()
+	require.NoError(t, err, "failed to set kubeconfig to source cluster: %v", err)
```
[the remark this answers is not in the document]
Comment (diptiranjanpx) [reply test/integration_test/migration_test.go]: done

File: test/integration_test/migration_test.go [hunk 8]
```diff
@@ -2027,6 +2032,288 @@ func transformResourceTest(t *testing.T) {
...
+
+	ctxs, preMigrationCtx := triggerMigration(
+		t,
+		instanceID,
+		appKey,
+		[]string{},
+		[]string{instanceID},
+		false,
+		false,
+		true,
```
[the remark this answers is not in the document]
Comment (diptiranjanpx) [reply test/integration_test/migration_test.go]: done

**Review by olavangad-px — APPROVED** [review]
LGTM

**Pull request merged**